autocomplete/builders/common.lua

```diff
@@ -393,54 +408,102 @@ end
 --- @field namespace string The full namespace of the package.
 --- @field deprecated boolean Allows marking definitions as deprecated. Those definitions aren't written to the web documentation.
 --- @field examples table<string, exampleTable>|nil A table containing the examples. Keys are the example's name/path to the example file.
 
+---@class packageValue : package
+---@field valuetype string
+---@field link string
+
+---@class packageFunctionArgument : package
+---@field tableParams packageFunctionArgument[]
+
+--- @class packageFunction : packageValue
+--- @field arguments packageFunctionArgument[]
+--- @field returns packageFunctionArgument[]|nil
+
+--- @class packageMethod : packageFunction
+
 --- @class packageLib : package
 --- @field children table<string, package>|nil
---- @field functions package[]|nil
---- @field values package[]|nil
+--- @field functions packageFunction[]|nil
+--- @field values packageValue[]|nil
+
+
+---@class packageOverload
+---@field rightType string
+---@field resultType string
+---@field description string
+
+---@class packageOperator : package
+---@field overloads packageOverload[]
 
 --- @class packageClass : packageLib
 --- @field inherits string The class that this class descends from.
+--- @field operators packageOperator[]
 --- @field isAbstract boolean
---- @field methods package[]|nil
+--- @field methods packageMethod[]|nil
 --- @field allDescendents table<string, packageClass>
 --- @field directDescendents table<string, packageClass>
+--- @field allDescendentKeys string
 
---- @class packageFunction : package
 
---- @class packageMethod : packageFunction
---- @field returns table
+
+---@class packageEvent : package
+---@field filter string
+---@field blockable boolean
+---@field related string[]
+---@field eventData table<string, packageFunctionArgument>
 
 --- comment
 --- @param package packageFunction
---- @return table
+--- @return packageFunctionArgument[]|nil
 function common.getConsistentReturnValues(package)
-	if (type(package.returns) == "string" and package.valuetype == nil) then
-		return { { name = "result", type = package.returns } }
-	elseif (type(package.returns) == "string" and package.valuetype ~= nil) then
+	if type(package.returns) == "string" then
+		if package.valuetype == nil then
+			return { { name = "result", type = package.returns } }
+		end
 		return { { name = package.returns, type = package.valuetype } }
-	elseif (package.valuetype) then
+	end
+	
+	if (package.valuetype) then
 		return { { name = "result", type = package.valuetype } }
-	elseif (type(package.returns) == "table") then
-		if (package.returns.name or package.returns.type) then
-			return { package.returns }
-		elseif (#package.returns > 0) then
-			return package.returns
-		end
-		error("Invalid parameters table.")
 	end
+
+	if type(package.returns) ~= "table" then return end
+
+	---@diagnostic disable-next-line: undefined-field
+	if (package.returns.name or package.returns.type) then
```

Comment:
Let's remove the `---@diagnostic disable-next-line: undefined-field` and add the `name` and `type` fields to the `packageFunctionArgument` annotation.

Reply:
I might be misunderstanding, but I don't think this is possible, since `package.returns` has type `packageFunctionArgument[]`, not `packageFunctionArgument`. I'm guessing that this is code to handle a special case when `package` is ""improperly"" initialized and `returns` has the ""wrong"" type (according to the annotations I added).

Maybe I could change the type of `returns` to:
```lua
packageFunctionArgument|packageFunctionArgument[]|nil
````
If I did that, I wouldn't need the `diagnostic` statement, since `packageFunctionArgument` extends `package` and inherits the relevant fields. 

But let me know what you think. I still don't have a very solid understanding of the inner workings of the documentation builder.

Reply:
Yeah, it's here to handle returns not being an array of `packageFunctionArgument`. It's fine to change it's type to `packageFunctionArgument|packageFunctionArgument[]|nil`